awx/sso/backends.py

```diff
@@ -374,90 +239,91 @@ def _update_m2m_from_groups(ldap_user, opts, remove=True):
-    team_map_settings = getattr(backend.settings, 'TEAM_MAP', {})
-    orgs_list = list(org_map.keys())
-    team_map = {}
-    for team_name, team_opts in team_map_settings.items():
-        if not team_opts.get('organization', None):
-            # You can't save the LDAP config in the UI w/o an org (or '' or null as the org) so if we somehow got this condition its an error
-            logger.error("Team named {} in LDAP team map settings is invalid due to missing organization".format(team_name))
-            continue
-        team_map[team_name] = team_opts['organization']
-
-    create_org_and_teams(orgs_list, team_map, 'LDAP')
-
-    # Compute in memory what the state is of the different LDAP orgs
-    org_roles_and_ldap_attributes = {'admin_role': 'admins', 'auditor_role': 'auditors', 'member_role': 'users'}
-    desired_org_states = {}
-    for org_name, org_opts in org_map.items():
-        remove = bool(org_opts.get('remove', True))
-        desired_org_states[org_name] = {}
-        for org_role_name in org_roles_and_ldap_attributes.keys():
-            ldap_name = org_roles_and_ldap_attributes[org_role_name]
-            opts = org_opts.get(ldap_name, None)
-            remove = bool(org_opts.get('remove_{}'.format(ldap_name), remove))
-            desired_org_states[org_name][org_role_name] = _update_m2m_from_groups(ldap_user, opts, remove)
-
-        # If everything returned None (because there was no configuration) we can remove this org from our map
-        # This will prevent us from loading the org in the next query
-        if all(desired_org_states[org_name][org_role_name] is None for org_role_name in org_roles_and_ldap_attributes.keys()):
-            del desired_org_states[org_name]
-
-    # Compute in memory what the state is of the different LDAP teams
-    desired_team_states = {}
-    for team_name, team_opts in team_map_settings.items():
-        if 'organization' not in team_opts:
-            continue
-        users_opts = team_opts.get('users', None)
-        remove = bool(team_opts.get('remove', True))
-        state = _update_m2m_from_groups(ldap_user, users_opts, remove)
-        if state is not None:
-            organization = team_opts['organization']
-            if organization not in desired_team_states:
-                desired_team_states[organization] = {}
-            desired_team_states[organization][team_name] = {'member_role': state}
-
-    # Check if user.profile is available, otherwise force user.save()
-    try:
-        _ = user.profile
-    except ValueError:
-        force_user_update = True
-    finally:
-        if force_user_update:
-            user.save()
-
-    # Update user profile to store LDAP DN.
-    profile = user.profile
-    if profile.ldap_dn != ldap_user.dn:
-        profile.ldap_dn = ldap_user.dn
-        profile.save()
-
-    reconcile_users_org_team_mappings(user, desired_org_states, desired_team_states, 'LDAP')
+# Do we still need to populate ldap field on profile
```

Comment:
I'm not sure if we want to leave this method in AWX as I didn't see a similar function in django-ansible-base to cover updating the Profile field.